woocommerce/woocommerce
# Fix empty last4 rendering

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR removes the extra `-` that was rendered on the receipt even when that `last4` card info was not available.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install `WooPayments` and enable sandbox mode.
2. In `wp-admin/admin.php?page=wc-settings&tab=checkout` make sure you have enabled the `Cash on delivery` payment method.
3. Make a purchase with a fake card and another one with `Cash on delivery`.
4. Go to `WooCommerce > Order` and note the numbers of the two orders you just did.
5. Go to `wp-admin/admin.php?page=wc-settings&tab=advanced&section=keys` and add a new key with `Read/Write` permissions.
6. Make the following for the two order numbers you got on step 4 request using the consumer key and secret generated in the previous step:
```bash 
curl --insecure -u [KEY]:[SECRET] --request POST --url 'https://[STORE_URL]/wp-json/wc/v3/orders/[ORDER_NUMBER]/receipt?force_new=true' --header 'Content-Type: application/json'
```
7. Visit the URL from the response and make sure the order made with the card shows the card icon, a `-`, and the last 4 digits of the card. Also, make sure the order made with `Cash on delivery` only shows an icons (no `-`).

| Cash | Card |
|--------|--------|
| <img width="218" alt="cash" src="https://github.com/user-attachments/assets/2ba0d01a-16c6-457f-aaf9-803f6b04decd">| <img width="216" alt="Screenshot 2024-10-28 at 14 50 59" src="https://github.com/user-attachments/assets/b4120bf6-b73f-4265-ba43-fdf0e40cd9bc"> | 




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Avoid rendering the dash in the receipts when there's no info about the card used to pay the order.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
